### PR TITLE
Port to meshkit and meshery-adapter-library

### PR DESCRIPTION
**Pull request by utkarsh-pro**
**Description**
This PR ports istio adapter to use meshkit and meshery-adapter-library. 

Functions tested (Both within and outside the container):
- [x] Installing/Uninstalling istio service mesh
- [x] Deploying sample applications (tested emojivoto)
- [x] Add/Remove label `istio-injection` to any namespace
- [x] Apply and Delete custom yamls
- [ ] SMI Conformance

**Notes for Reviewers**
SMI Conformance test is not working at the moment, likely an issue in the meshery-adapter-library.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@utkarsh-pro I missed what the .exe consideration is here.

**Comment by utkarsh-pro** [comment]
> @utkarsh-pro I missed what the .exe consideration is here.

@leecalcote, initially I was appending ".exe" at the end of the istioctl binary (for windows platofrm) assuming that windows binaries need to have that extension. It would have created a bit of issue at other places also. But then I tried a simple go program (in windows system) to execute binary without "exe" extension and it worked just fine so I removed that additional code snippet which was adding the ".exe" extension at the end of binary name on the windows os.

Here is the code snipped that I tried on windows:
```go
package main

import (
	"bytes"
	"fmt"
	"log"
	"os/exec"
)

func main() {
	var (
		out bytes.Buffer
		er  bytes.Buffer
	)
	Executable := `C:\Users\sriva\Downloads\istioctl-1.8.0-win\istioctl`
	command := exec.Command(Executable)
	command.Stdout = &out
	command.Stderr = &er
	err := command.Run()
	if err != nil {
		log.Fatalln(err)
	}

	fmt.Println(out.String())
}
```

**Comment by utkarsh-pro** [comment]
> @utkarsh-pro I missed what the .exe consideration is here.

@leecalcote, initially I was appending ".exe" to the istioctl binary name while renaming the binaries but then I read up that ".exe" extension is not really necessary in windows too (just like in UNIX based os). 
But I have just tested it out on a windows system, although it is not necessary to have ".exe" extension name in the name of executable (within a go program) it is mandatory to have it in the file system. Making the necessary changes now and pushing the updates.

Tested on Windows 10

File: internal/config/config.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+	"github.com/layer5io/meshery-adapter-library/common"
+	"github.com/layer5io/meshery-adapter-library/config"
+	configprovider "github.com/layer5io/meshery-adapter-library/config/provider"
+	"github.com/layer5io/meshery-adapter-library/status"
+	"github.com/layer5io/meshkit/utils"
+)
+
+const (
+	IstioOperation = "istio"
```
Comment (leecalcote) [inline internal/config/config.go:14]:
```suggestion
	// Constants to use in log statements
	IstioOperation = "istio"
```

File: internal/config/operations.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		Versions:    versions,
+		Templates: []adapter.Template{
+			"templates/istio.yaml",
+		},
+		AdditionalProperties: map[string]string{},
+	}
+
+	dev[LabelNamespace] = &adapter.Operation{
+		Type:        int32(meshes.OpCategory_CONFIGURE),
+		Description: "Label Namespace",
```
Comment (leecalcote) [inline internal/config/operations.go:27]:
```suggestion
		Description: "Label Namespace for Automatic Sidecar Injection",
```

File: istio/custom_operations.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package istio
```
Comment (leecalcote) [inline istio/custom_operations.go:1]:
```suggestion
// Common operations for the adapter
package istio
```

File: istio/error.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
+package istio
```
Comment (leecalcote) [inline istio/error.go:1]:
```suggestion
// Error codes for the adapter
package istio
```

File: istio/install.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+	var (
+		out bytes.Buffer
+		er  bytes.Buffer
+	)
+
+	Executable, err := istio.getExecutable(version)
+	if err != nil {
+		return "", ErrFetchManifest(err, err.Error())
+	}
+	execCmd := []string{"install", "--set", "profile=demo", "-y"}
```
Comment (leecalcote) [inline istio/install.go:74]: Future: allow user to provide profile name.

**Review by leecalcote — APPROVED** [review]

File: istio/error.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
+// Error codes for the adapter
```
Comment (leecalcote) [inline istio/error.go:1]:
```suggestion
// Package istio - Error codes for the adapter
```

File: istio/error.go [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+package istio
+
+import (
+	"fmt"
+
+	"github.com/layer5io/meshkit/errors"
+)
+
+var (
+	ErrInstallIstioCode    = "istio_test_code"
```
Comment (leecalcote) [inline istio/error.go:11]:
```suggestion
	// Errror code for failed service mesh installation
	ErrInstallIstioCode    = "istio_test_code"
```

File: istio/custom_operations.go [hunk 8]
```diff
@@ -0,0 +1,17 @@
+// Common operations for the adapter
```
Comment (leecalcote) [inline istio/custom_operations.go:1]:
```suggestion
// Package istio - Common operations for the adapter
```

File: istio/error.go (hunk 6 above)
Comment (leecalcote) [inline istio/error.go:1]:
```suggestion
// Package istio - Error codes for the adapter
```

**Review by leecalcote — APPROVED** [review]

**Pull request merged**